kernel/cpu_mask.c

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) 2018,2024 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <zephyr/kernel.h>
+#include <ksched.h>
+#include <zephyr/spinlock.h>
+
+extern struct k_spinlock sched_spinlock;
+
+
+# ifdef CONFIG_SMP
+/* Right now we use a two byte for this mask */
+BUILD_ASSERT(CONFIG_MP_MAX_NUM_CPUS <= 16, "Too many CPUs for mask word");
```

Comment:
correct, but an architecture can redefine this? like in arch/x86/core/Kconfig.intel64

you see:

```
config MP_MAX_NUM_CPUS
        range 1 4
```

This is lowering the number, but someone might also go over 12. Anyways, I am just moving code around, this can be dealt with in a followup